mantidproject/mantid
# Create algorithm SANSILLParameterScan

**Description of work.**
This PR implements the first part of the parameter scan algorithm. Given a series of files from D16, it does the usual corrections, then sum each vertical pixel column for each file, resulting in 1D workspaces, and then concatenates them sorted by the parameter scanned - usually omega.
The output is thus a 2D workspace, with X the scattering angle and Y the parameter. 


**To test:**
 - Run 
```
from mantid.simpleapi import *

config.appendDataSearchSubDir('ILL/D16/')
config.setFacility("ILL")
SANSILLParameterScan(SampleRuns="023583:023585",
                     Output2D="output2d",
                     SampleThickness=0.1,
                     Observable="Omega.value",
                     PixelYmin=3,
                     PixelYMax=317)
```
- Open `output2d` in colorfill mode, and check it shows a vertical peak line near 5 degrees of scattering angle. 



Fixes #27615. 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
